test/index.js

```diff
@@ -1,196 +1,178 @@
+      done();
+  });
+
+  it('should support instantFlush', function(done) {
+      var statsCollector = Collector(COLLECTOR_URL + '?instantFlush=yes');
+      var writePointsSpy = simple.mock(statsCollector._client, 'writePoints');
+
+      statsCollector.collect('series-name', {foo: 34.0});
+
+      assert.equal(writePointsSpy.callCount, 1);
+      var writeArgs = [
+        [
+          {
+            measurement: 'series-name',
+            tags: undefined,
+            fields: {foo: 34.0}
+          }
+        ],
+        {precision: undefined}
+      ];
+      assert.deepEqual(writePointsSpy.calls[0].args, writeArgs);
+      done();
+  });
+
+  it('should support a custom flushInterval', function(done) {
+      var statsCollector = Collector(COLLECTOR_URL + '?flushInterval=500');
+      var writePointsSpy = simple.mock(statsCollector._client, 'writePoints');
+
+      statsCollector.collect('series-name', {foo: 34.0});
+
+      assert.equal(writePointsSpy.callCount, 0);
+      setTimeout(function() {
+        assert.equal(writePointsSpy.callCount, 1);
+        var writeArgs = [
+          [
+            {
+              measurement: 'series-name',
+              tags: undefined,
+              fields: {foo: 34.0}
+            }
+          ],
+          {precision: undefined}
+        ];
+        assert.deepEqual(writePointsSpy.calls[0].args, writeArgs);
         done();
-    }, 1000);
-});
-
-test('should support a autoFlush=no', function(done) {
-    var stats = Collector(COLLECTOR_URL + '?flushInterval=500&autoFlush=no');
-
-    var req = nock('http://example.com:8086')
-    .filteringRequestBody(/.*/, function(body) {
-        return '*';
-    })
-    .post('/write', '*')
-    .query({db: 'test-db', u: 'user', p: 'password', precision: 'ms', database: 'test-db' })
-    .reply(200, function(uri, request_body) {
-        assert.equal(request_body, 'series foo=34');
-    });
-
-    stats.collect('series', { foo: 34.0 })
-
-    setTimeout(function() {
-        assert(req.isDone() == false);
-
-        stats.flush();
-
-        setTimeout(function() {
-            req.done();
-            done();
-        }, 200);
-    }, 1000);
-});
-
-test('should support time_precision', function(done) {
-    var stats = Collector(COLLECTOR_URL + '?time_precision=s');
-
-    var req = nock('http://example.com:8086')
-    .filteringRequestBody(/.*/, function(body) {
-        return '*';
-    })
-    .post('/write', '*')
-    .query({db: 'test-db', u: 'user', p: 'password', precision: 's', database: 'test-db' })
-    .reply(200, function(uri, request_body) {
-        assert.equal(request_body, 'series foo=34 1416512521');
-    });
-
-    stats.collect('series', { time: 1416512521, foo: 34.0 });
-
-    stats.flush();
-    setTimeout(function() {
-        req.done();
+      }, 600);
+  });
+
+  it('should support a autoFlush=no', function(done) {
+      var statsCollector = Collector(COLLECTOR_URL + '?flushInterval=500&autoFlush=no');
```

Comment:
`flush` eventually calls the underlying `node-influx` library's `writePoints`.

The test is effectively checking that this doesn't flush until `.flush()` is called by asserting that `writePoints` function isn't called even after waiting beyond the flush interval.